### PR TITLE
Add setup file and prepare for publish in pip

### PR DESCRIPTION
**Pull request by superstar54**
This PR supports installing `batoms` as a Python module, thus use it with `bpy` in any Python script, with access to Blender’s features.

Check this [wiki](https://github.com/beautiful-atoms/beautiful-atoms/wiki/Use-Blender-as-a-Python-module)

### CODE REVIEW

**Comment by alchem0x2A** [comment]
That's a huge improvement! I didn't know the official bpy build before (there was a placeholder "bpy" project at https://pypi.org/project/fake-bpy-module-latest/, but not straightforward). I assume this means we can basically 1) deprecate batoms-api, 2) simplify the installer script right?


The only apparent problem is bpy as a module requires the exact same python version as in build (Python ==3.10.*) I'm not sure how strict will the requirement for minor version be, but it seems at least the major version should match.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    url='https://github.com/beautiful-atoms/beautiful-atoms',
+    install_requires=[
+        "bpy",
+        "ase",
+    ],
+    classifiers=[
+        "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+        'Programming Language :: Python :: 3',
+    ],
+    python_requires='>=3.10',
```
Comment (alchem0x2A) [inline setup.py:22]: per pypi description, better to put "==3.10"? https://pypi.org/project/bpy/3.6.0/#files

**Comment by alchem0x2A** [comment]
on the CI testings, macos builders seems to have really weird issues. we can ignore them now

**Comment by alchem0x2A** [comment]
on the other hand, how does the pip installation affect the installer script? I tested on a blender3.5+batoms environment and installed bpy 3.6, and they seem to not interfere with each other. Maybe this means we can include both bpy / batoms via pip inside the installer script, just to make sure the versions are correctedly handled.

**Comment by superstar54** [comment]
I tested using a different minor Python version (3.10.0) for bpy3.6.0, and all the basic tests (atoms, bond, species, boundary) passed. It seems that the minor version doesn't matter.

However, the major version is still a restriction, which may prevent its use in other workflows, so we may keep the old batoms-api.

For the installation, we could add a new installation option using pip only, especially for HPC.

**Pull request merged**